src/api/api-categories/api-categories.ts

```diff
@@ -0,0 +1,67 @@
+import axios from "axios";
+import { axiosConfig } from "../../state/state";
+
+export interface ICategory {
+  parentCategoryId: string;
+  name: string;
+  description: string;
+  icon: string;
+  imagePath: string;
+}
+export interface ICategories {
+  name: string;
+  description: string;
+  subcategories: [];
+  uid: string;
+}
+
+export interface ICategoriesQuantity {
+  quantity: number;
+}
+
+export class ApiCategories {
+  apiUrl: string;
+
+  constructor(apiUrl: string) {
+    this.apiUrl = apiUrl;
+  }
+
+  public async add(categoriesObj: ICategory): Promise<ICategory> {
+    const apiUrl: string = this.apiUrl + "categories";
+
+    try {
+      const response = await axios.post<ICategory>(apiUrl, null, {
+        params: categoriesObj,
+        ...axiosConfig,
+      });
+      return response.data;
+    } catch (error) {
+      console.log(error);
+      throw new Error("Api add category error");
+    }
+  }
+  public async get(): Promise<ICategories[]> {
```

Comment:
Be consist, if you're name next function like `getProductQuantity then maybe instead of `get` name this one as getProductCategories ?